### PR TITLE
Add option to init config from string

### PR DESCRIPTION
**Pull request by pronebird**
Hi,

First of all thank you for writing that beautiful piece of software.

This PR adds ability to provide a config via byte array, right now I am simply passing utf8 string from Rust and that seems to work just fine.

I am not entirely sure that the formal way to add support for that is to add more args to `main()` so feel free to refine that PR if you find it useful.

### CODE REVIEW

File: src/hev-main.h [hunk 1]
```diff
@@ -28,7 +28,7 @@ extern "C" {
  *
  * Since: 2.4.6
  */
-int hev_socks5_tunnel_main (const char *config_path, int tun_fd);
+int hev_socks5_tunnel_main (const char *config_path, const unsigned char *config_str, unsigned int config_len, int tun_fd);
```
Comment (heiher) [inline src/hev-main.h:31]: Could you add a new method/function to avoid break the current API? Thanks.

Comment (pronebird) [reply src/hev-main.h]: @heiher added `hev_socks5_tunnel_main2()` awkward name but it's either long or that.. :)

Comment (heiher) [reply src/hev-main.h]: Thanks.

**Review by heiher — APPROVED** [review]
Thanks.

**Pull request merged**